src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -746,25 +748,50 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList,
 		char *shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
 		char *quotedShardName = quote_literal_cstr(shardQualifiedName);
 
+		/* for partitoned tables, call worker_partitioned_... size functions */
 		if (optimizePartitionCalculations && PartitionedTable(shardInterval->relationId))
 		{
 			appendStringInfo(selectQuery, GetWorkerPartitionedSizeUDFNameBySizeQueryType(
 								 sizeQueryType), quotedShardName);
+			appendStringInfo(selectQuery, " + ");
```

Comment:
I feel like this + is going to become an issue at some point in the future too. If someone has lots of colocated partitioned tables. So let's make this use the VALUES list + SUM trick too.